### PR TITLE
Week2 hw

### PR DESCRIPTION
**Pull request by EsraZengn**
I completed and uploaded the remaining parts of my homework. Thank you very much for feedback of  the first part, looking forward to the remaining ones.

### CODE REVIEW

File: Week2/homework/index.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+      }
+      exitProgram();
+    } catch (error) {
+      console.error(`There was an error: ${error.code} - ${error.message}`);
+    }
+  }
+
+  async function findCapitalCity() {
+    const country = readline.question('Enter a country: ');
+    const query = `select ci.name as capital from city ci inner join country co on ci.id = co.capital where co.name = ? `;
```
Comment (mogaleaf) [inline Week2/homework/index.js]: good

File: Week2/homework/index.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    await makeQuery(query, country);
+  }
+
+  async function listLanguages() {
+    const region = readline.question('Enter a region: ');
+    const query = `select distinct cl.language from countrylanguage cl inner join country c on cl.countryCode = c.code where c.region = ?`;
```
Comment (mogaleaf) [inline Week2/homework/index.js]: good

File: Week2/homework/index.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+  async function findCountries() {
+    try {
+      const region = readline.question('Enter a region: ');
+      const language = readline.question('Enter a language: ');
+      const countries = await executeQuery(
+        `select distinct c.name as countries from countrylanguage l 
+          inner join (select * from country where region = ? ) as c 
+          on c.code = l.countryCode 
+          and l.language = ? 
+          and l.isOfficial = 't'`,
```
Comment (mogaleaf) [inline Week2/homework/index.js]: Yes it's good but I would have done :
select distinct c.name as countries from countrylanguage l 
          inner join country as c 
          on c.code = l.countryCode
          where
          c.region = ? 
          and l.language = ? 
          and l.isOfficial = 't'` ;

We usually join on the table and use it in the where clause

File: Week2/homework/index.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+      exitProgram();
+    } catch (error) {
+      console.error(
+        `There was error while calculating the countries: ${error.code} - ${error.message}`,
+      );
+    }
+  }
+
+  async function listContinents() {
+    const query = `select c.continent, count( distinct l.language) as numberOfLanguagesSpoken from country c
```
Comment (mogaleaf) [inline Week2/homework/index.js]: good !

File: Week2/homework/index.js [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    const query = `select distinct cl.language from countrylanguage cl inner join country c on cl.countryCode = c.code where c.region = ?`;
+    console.log(`The languages spoken in ${region}: `);
+    await makeQuery(query, region);
+  }
+
+  async function findNumberOfCities() {
+    try {
+      const language = readline.question('Enter a language:');
+      const result = await executeQuery(
+        `select count(distinct ci.name) as numberOfCities from city ci
```
Comment (mogaleaf) [inline Week2/homework/index.js]: good

File: Week2/homework/index.js (hunk 1 above)
Comment (EsraZengn) [reply Week2/homework/index.js]: Thank you.

File: Week2/homework/index.js (hunk 2 above)
Comment (EsraZengn) [reply Week2/homework/index.js]: Thank you.

File: Week2/homework/index.js (hunk 3 above)
Comment (EsraZengn) [reply Week2/homework/index.js]: I see your point, thanks for the explanation I will change mine as you suggested :+1:

File: Week2/homework/index.js (hunk 4 above)
Comment (EsraZengn) [reply Week2/homework/index.js]: thanks :)

File: Week2/homework/index.js (hunk 5 above)
Comment (EsraZengn) [reply Week2/homework/index.js]: thanks

**Comment by EsraZengn** [comment]
Thank you very much for your feedbacks.

**Pull request closed without merging**